internal/handlers/llm_mcp_bridge.go

```diff
@@ -224,54 +224,62 @@ func NewLLMMCPBridgeFromClientsWithLogLevel(mcpClients interface{}, stdLogger *l
 	return NewLLMMCPBridgeWithLogLevel(interfaceClients, stdLogger, discoveredTools, logLevel, llmRegistry, cfg)
 }
 
-// ProcessLLMResponse processes an LLM response, expecting a specific JSON tool call format.
-// It no longer uses natural language detection.
-func (b *LLMMCPBridge) ProcessLLMResponse(ctx context.Context, llmResponse *llms.ContentChoice, _ string, extraArgs map[string]interface{}) (string, error) {
+// ExtractToolCall extracts tool call information from LLM response
+// Returns nil if no tool call is detected
+func (b *LLMMCPBridge) ExtractToolCall(llmResponse *llms.ContentChoice) (*ToolCall, error) {
```

Comment:
ProcessLLMResponse removed, split into ExtractToolCall + ExecuteToolCall. No deprecation path.

The error return changed from `(string, nil)` to `("", error)` - callers expecting old behavior will break.

```
// OLD: return errorMessage, nil
// NEW: return "", fmt.Errorf(errorMessage
```